### PR TITLE
Build with Numpy 2.0, remove setup.cfg and more

### PR DESCRIPTION
**Pull request by saimn**
@cmccully - with Numpy 2.0 coming soon we will need to publish new wheels built with Numpy 2.0 (so they will be compatible with both Numpy 1.x and 2.0).

Also did various infrastructure updates to get rid of warnings etc.

### CODE REVIEW

File: astroscrappy/utils/setup_package.py [hunk 1]
```diff
@@ -1,39 +1,36 @@
...
+    im_sources = [os.path.join(UTIL_DIR, "image_utils.pyx"),
+                  os.path.join(UTIL_DIR, "imutils.c")]
 
     include_dirs = [np.get_include(), UTIL_DIR]
-    libraries = []
 
     if 'CFLAGS' in os.environ:
         extra_compile_args = os.environ['CFLAGS'].split()
     else:
         extra_compile_args = ['-g', '-O3', '-funroll-loops', '-ffast-math']
```
Comment (saimn) [inline astroscrappy/utils/setup_package.py:23]: @cmccully - Not sure if we should have those flags by default, especially `-ffast-math` which is an option that people should use carefully (https://simonbyrne.github.io/notes/fastmath/)

**Comment by saimn** [comment]
`test_median_clean` is failing when running tests for the cp311-win_amd64 wheel... 
Not the first time we have a problem with this test (#71, #82, https://github.com/astropy/astroscrappy/pull/66#issuecomment-973287545) so probably something that should be investigated more in detail, for now I will just mark it as xfail.
```
      def test_median_clean(testdata):
          imdata, crmask = testdata
          # Because our image only contains single cosmics, turn off
          # neighbor detection. Also, our cosmic rays are high enough
          # contrast that we can turn our detection threshold up.
          _mask, clean = detect_cosmics(imdata, readnoise=10., gain=1.0,
                                        sigclip=6, sigfrac=1.0, cleantype='median')
      
          assert (clean[crmask] != imdata[crmask]).sum() == crmask.sum()
      
          # Run it again on the clean data. We shouldn't find any new cosmic rays
          _mask2, _clean2 = detect_cosmics(clean, readnoise=10., gain=1.0,
                                           sigclip=6, sigfrac=1.0, cleantype='median')
  >       assert _mask2.sum() == 0
  E       assert 100 == 0
```

**Comment by saimn** [comment]
@cmccully - PR is ready, do you want to have a look ?

**Review by astrofrog — APPROVED** [review]
This looks good to me - I think we should also switch to the OpenAstronomy wheel building workflow but I'm happy to do that in a follow-up PR.

**Comment by astrofrog** [comment]
As this is mainly infrastructural, I think we should go ahead and merge so that other packages can work properly with Numpy 2.0. However, I don't have permission to merge, so need to find someone who does.

**Pull request merged**